hammerblade/torch/tests/test_log_softmax_backward.py

```diff
@@ -0,0 +1,45 @@
+"""
+Tests on torch.nn.LogSoftMax backward
+03/26/2020 Lin Cheng (lc873@cornell.edu)
+"""
+
+import torch
+import torch.nn as nn
+import numpy as np
+
+def test_torch_nn_LogSoftMax_back_1():
+    m = nn.LogSoftmax(dim=1)
+    loss = nn.NLLLoss(reduction='mean')
+    input = torch.tensor([[-3.5164, -1.9673, -4.7514, -0.2075, -4.6912],
+                         [-2.9321, -1.5085, -0.8699, -1.4016, -2.8089],
+                         [-2.6460, -1.9800, -1.4818, -1.8358, -0.9057]],
+                         requires_grad=True)
+    input_h = torch.tensor([[-3.5164, -1.9673, -4.7514, -0.2075, -4.6912],
+                           [-2.9321, -1.5085, -0.8699, -1.4016, -2.8089],
+                           [-2.6460, -1.9800, -1.4818, -1.8358, -0.9057]],
+                           requires_grad=True)
+    target = torch.tensor([1, 0, 4])
+    output = loss(m(input), target)
+    logsoftmax = m(input_h.hammerblade())
+    assert logsoftmax.device == torch.device("hammerblade")
+    output_h = loss(logsoftmax.cpu(), target)
```

Comment:
Is it required to do backward on `loss` and not directly `LogSfotmax` to test `log_softmax_backward`? I'd imagine later would be faster test...?

Reply:
Yes, technically we can. I was just lazy and reused old tests (with different parts on cpu) XD

Reply:
Updated to directly test logsoftmax_backward! Maybe take a look and see if there is anything wrong?

Reply:
Looks great! Any advantage with `np.random` over `torch.rand`? (I'll try to use that in my tests if so...)

Reply:
I think the reason is just I want to create two identical tensors (without clone etc). I don't think one is better than the other ;)

Reply:
Got it.